app/assets/javascripts/home.js.erb

```diff
@@ -1,4 +1,60 @@
 var map;
+var currentTarget;
+var existsCurrentTarget = false;
+
+$(document).ready(function(){
+
+  // Only numbers in radius field validation.
+  $("#index-dynamic-container").on('keydown', "#target_radius", function (e) {
+    // Allow: backspace, delete, tab, escape, enter and .
```

Comment:
you could put this validation in a function, probably in application.js and call it from there.